Tests/FakeFunctionTests.class.sql

```diff
@@ -422,6 +431,131 @@ BEGIN
   EXEC FakeFunctionTests.[assert parameter missmatch causes error];
 END;
 GO
+CREATE PROCEDURE FakeFunctionTests.[test errors when both @FakeFunctionName and @FakeDataSource are passed]
+AS
+BEGIN
+  EXEC('CREATE FUNCTION FakeFunctionTests.AFunction(@a int, @b int) RETURNS TABLE AS RETURN (SELECT @a AS one);');
+  EXEC tSQLt.ExpectException @ExpectedMessage = 'Either @FakeFunctionName or @FakeDataSource must be provided', @ExpectedSeverity = 16, @ExpectedState = 10;
+  EXEC tSQLt.FakeFunction @FunctionName = 'FakeFunctionTests.AFunction';
+END;
+GO
+CREATE PROCEDURE FakeFunctionTests.[test errors when neither @FakeFunctionName nor @FakeDataSource are passed]
+AS
+BEGIN
+  EXEC('CREATE FUNCTION FakeFunctionTests.AFunction(@a int, @b int) RETURNS TABLE AS RETURN (SELECT @a AS one);');
+
+  EXEC tSQLt.ExpectException @ExpectedMessage = 'Both @FakeFunctionName and @FakeDataSource are valued. Please use only one.', @ExpectedSeverity = 16, @ExpectedState = 10;
+
+  EXEC tSQLt.FakeFunction @FunctionName = 'FakeFunctionTests.AFunction', @FakeFunctionName = 'FakeFunctionTests.AFunction', @FakeDataSource = 'select 1 one';
+END;
+GO
+CREATE PROCEDURE FakeFunctionTests.[test errors when scalar function is passed as @FakeDataSource]

```

Comment:
The test name is unclear. How about [test errors if function is a SVF and @FakeDataSource is used]?

In addition: We need a similar test for CLR SVF. I did not see that.